src/components/tx-flow/common/TxNonce/index.tsx

```diff
@@ -109,6 +109,16 @@ const TxNonceForm = ({ nonce, recommendedNonce }: { nonce: string; recommendedNo
     formMethods.setValue(TxNonceFormFieldNames.NONCE, recommendedNonce)
   }
 
+  const currentFormNonce = formMethods.watch(TxNonceFormFieldNames.NONCE)
+  const isFormChanged = formMethods.formState.isDirty
+
+  // Update to recommended nonce if the current nonce is unchanged and invalid after update
+  useEffect(() => {
+    if (!isFormChanged && Number(currentFormNonce) < safe.nonce) {
```

Comment:
Why are we checking the safe.nonce here and not the recommended nonce?

Reply:
This fix is mainly for the case when a user immediately executes, prepares a second tx while the first is indexing. Now with this change we will update the nonce to the next one once the indexer catches up. Previously we left an invalid nonce and did not show any error.

Reply:
Yeah, but you're setting the input value to recommendedNonce, so you expect it to be updated. Then why not compare against that?

Reply:
True. I will change it to reset to the `safe.nonce` as that is updated / correct for sure. I think comparing to the `recommendedNonce` could sometimes overwrite a lower nonce for instance when replacing a transaction.

Reply:
OK. Another question:

When you call setValue like this, what happens in the SafeTxProvider? Will it trigger a setNonce call? Which would be an equivalent of manually setting the nonce? Will this effectively stop it from being further updated should a new recommendedNonce arrive in the meanwhile?